src/plugins/plugin.legend.js

```diff
@@ -495,21 +502,33 @@ var Legend = Element.extend({
 
 				if (x >= hitBox.left && x <= hitBox.left + hitBox.width && y >= hitBox.top && y <= hitBox.top + hitBox.height) {
 					// Touching an element
+					hoveredItem = me.legendItems[i];
+
 					if (type === 'click') {
 						// use e.native for backwards compatibility
 						opts.onClick.call(me, e.native, me.legendItems[i]);
 						changed = true;
 						break;
 					} else if (type === 'mousemove') {
-						// use e.native for backwards compatibility
-						opts.onHover.call(me, e.native, me.legendItems[i]);
-						changed = true;
+						if (opts.onHover) {
```

Comment:
This can be combined to previous line to prevent codeclimate error
`} else if (type === 'mousemove' && opts.onHover) {`

Reply:
That wouldn't be equivalent. Right now the code does `break` if only `mousemove` occurred

Reply:
I could remove both `break`s from the the `if` blocks and have a single one afterwards?

Reply:
It looks like that would work

If we're in the mood to cleanup this code, I'd also probably suggest deleting the `changed` variable by replacing each `changed = true; break;` with `return true;` and then `return false;` on the very last line of the method

Reply:
What is the purpose of `changed` / the return value of `handleEvent`? It is ignored by `afterEvent`, and that appears to be the only place it is called.

I have an idea for a slightly larger refactor that might improve this method. I'll update the PR with it.

Reply:
I think you should still return `true` or `false` for backwards compatibility in case someone has built a plugin or chart that relies on it

Reply:
OK I'll add that back in

Reply:
`changed` was [legacy code](/chartjs/Chart.js/pull/3718/files#diff-de7249441169d0e646c53d8220e8b0c5L754) before refactoring the legend as plugin. It was used to tell the controller that something changed and a new render was required. But `onClick` calls `update()`, making this returned value from `handleEvent`  useless. Note that the [calling code disregard the returned value](/chartjs/Chart.js/blob/master/src/plugins/plugin.legend.js#L570), so it should be fine to remove that logic completely. (@etimberg can you confirm?)

Reply:
I've fixed the other issues, but waiting on confirmation about what to do here? Reintroduce the `changed` logic or leave it out?